presto-clp/src/main/java/com/facebook/presto/plugin/clp/ClpFilterToKqlConverter.java

```diff
@@ -0,0 +1,738 @@
+     *
+     * @param node the NOT call expression
+     * @return a ClpExpression with the translated KQL query, or the original expression if unsupported
+     */
+    private ClpExpression handleNot(CallExpression node)
+    {
+        if (node.getArguments().size() != 1) {
+            throw new PrestoException(CLP_PUSHDOWN_UNSUPPORTED_EXPRESSION,
+                    "NOT operator must have exactly one argument. Received: " + node);
+        }
+
+        RowExpression input = node.getArguments().get(0);
+        ClpExpression expression = input.accept(this, null);
+        if (expression.getRemainingExpression().isPresent() || !expression.getDefinition().isPresent()) {
+            return new ClpExpression(node);
+        }
+        return new ClpExpression("NOT " + expression.getDefinition().get());
+    }
+
+    /**
+     * Handles LIKE expressions.
+     * Converts SQL LIKE patterns into equivalent KQL queries using * (for %) and ? (for _).
+     * Only supports constant patterns or constant cast patterns.
+     * Example:
+     * - col1 LIKE 'a_bc%' → col1: "a?bc*"
+     *
+     * @param node the LIKE call expression
+     * @return a ClpExpression with the KQL equivalent of the LIKE expression, or the original node if unsupported
+     */
+    private ClpExpression handleLike(CallExpression node)
+    {
+        if (node.getArguments().size() != 2) {
+            throw new PrestoException(CLP_PUSHDOWN_UNSUPPORTED_EXPRESSION, "LIKE operator must have exactly two arguments. Received: " + node);
+        }
+        ClpExpression variable = node.getArguments().get(0).accept(this, null);
+        if (!variable.getDefinition().isPresent()) {
+            return new ClpExpression(node);
+        }
+
+        String variableName = variable.getDefinition().get();
+        RowExpression argument = node.getArguments().get(1);
+
+        String pattern;
+        if (argument instanceof ConstantExpression) {
+            ConstantExpression literal = (ConstantExpression) argument;
+            pattern = getLiteralString(literal);
+        }
+        else if (argument instanceof CallExpression) {
+            CallExpression callExpression = (CallExpression) argument;
+            if (!standardFunctionResolution.isCastFunction(callExpression.getFunctionHandle())) {
+                return new ClpExpression(node);
+            }
+            if (callExpression.getArguments().size() != 1) {
+                throw new PrestoException(CLP_PUSHDOWN_UNSUPPORTED_EXPRESSION, "CAST function must have exactly one argument. Received: " + callExpression);
+            }
+            if (!(callExpression.getArguments().get(0) instanceof ConstantExpression)) {
+                return new ClpExpression(node);
+            }
+            pattern = getLiteralString((ConstantExpression) callExpression.getArguments().get(0));
+        }
+        else {
+            return new ClpExpression(node);
+        }
+        pattern = pattern.replace("%", "*").replace("_", "?");
```

Comment:
Do we handle escapes? (E.g., `col LIKE 'abc#%def' ESCAPE '#'` --> `col: "abc%def"`

Reply:
No. It's `Likepattern` predicate, not `like`